opengrok-web/src/main/java/org/opengrok/web/api/v1/controller/RepositoriesController.java

```diff
@@ -24,28 +24,45 @@
 
 import org.opengrok.indexer.configuration.RuntimeEnvironment;
 import org.opengrok.indexer.history.RepositoryInfo;
+import org.opengrok.indexer.util.ClassUtil;
 
 import javax.ws.rs.GET;
 import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
 import javax.ws.rs.Produces;
 import javax.ws.rs.QueryParam;
+import javax.ws.rs.WebApplicationException;
 import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.io.IOException;
 
 @Path("/repositories")
 public class RepositoriesController {
 
     private RuntimeEnvironment env = RuntimeEnvironment.getInstance();
 
-    @GET
-    @Path("/type")
-    @Produces(MediaType.TEXT_PLAIN)
-    public String getType(@QueryParam("repository") final String repositoryPath) {
+    private RepositoryInfo getRepositoryInfo(String repositoryPath) {
         for (RepositoryInfo ri : env.getRepositories()) {
             if (ri.getDirectoryNameRelative().equals(repositoryPath)) {
-                return repositoryPath + ":" + ri.getType();
+                return ri;
             }
         }
-        return repositoryPath + ":N/A";
+
+        return null;
     }
 
+    @GET
+    @Path("/property/{field}")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Object get(@QueryParam("repository") final String repositoryPath, @PathParam("field") final String field)
+            throws IOException {
+
+        RepositoryInfo ri = getRepositoryInfo(repositoryPath);
+        if (ri == null) {
+            throw new WebApplicationException("cannot find repository with path: " + repositoryPath,
+                    Response.Status.NOT_FOUND);
+        }
+
+        return ClassUtil.getFieldValue(ri, field);
```

Comment:
I object to exposing data got via encapsulation-breaking reflection through web APIs like this. It was done for `Configuration`, and I would hope to see it undone.

The safer way to my thinking is to create a transfer object with public fields where an OpenGrok developer has to explicitly determine what is safely copied out of `RepositoryInfo` (or `Configuration`) to be exposed via `getFieldValue` through a web API.

Reply:
Same thing is done also in `ProjectsController`. I realize this is a bit of a lazy approach, wonder if there is a way how to create/generate such object using annotations.

Reply:
Anyway, added DTO sub-class just to see how it would look like. Is this something you had in mind ?

Reply:
Also, I see this approach should fix #2917.

Reply:
Using bean generator from cglib works fine. I like it better than duplicated getters/setters. This approach can be pushed further - collection of name/type pairs, annotations, etc.